iceoryx2/src/service/attribute.rs

```diff
@@ -126,9 +125,10 @@ pub struct Attribute {
 impl Attribute {
     /// Creates an attribute instance
     pub fn new(key: &str, value: &str) -> Self {
```

Comment:
We need to adjust the API here and use `pub fn new(key: &AttributeKeyString, value: &AttributeValueString) -> Self {`

and the user can use it like:

```rust
Attribute::new("hello".try_into()?, "world".try_into()?)
```

Otherwise a user input of a longer string would lead to a fatal crash which is a very bad idea.